wp-parsely.php

```diff
@@ -26,7 +26,9 @@
 
 if ( class_exists( 'Parsely' ) ) {
 	define( 'PARSELY_VERSION', Parsely::VERSION );
-	define( 'PARSELY_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+	if ( ! defined( 'PARSELY_PLUGIN_DIR' ) ) {
+		define( 'PARSELY_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+	}
```

Comment:
Why does it need a `defined()` check? What benefit is there for someone to define this constant to any other value?

Also, why does it need to only be defined if the Parsely class exists? We know the class exists, as we just required the file that defines it.

Reply:
> Why does it need a defined() check?

If we try to define an already defined constant, it's a fatal.  This is a hedge against hosts (including us) who may need to do funky things with file systems or otherwise have this constant set.

Reply:
> why does it need to only be defined if the Parsely class exists?

It really doesn't. I was just trying to keep all the `define` calls together and the `PARSELY_VERSION` one was already wrapped in that conditional.  I'll go ahead and pull them all out.

Reply:
> This is to prevent the notice for folks who otherwise have this constant set.

No-one will have this set - we've only just started using it. And the chance of someone using it for some other purpose is highly mitigated by the fact it has a prefix that matches the plugin name.

In this case, the constant is a convenience thing for us - we should take ownership of it.